### PR TITLE
Adding log support

### PR DESCRIPTION
**Pull request by jkurz**
- simple logging support

More features can be implemented later. This is basic support for logs with harbor-compose.

### CODE REVIEW

File: cmd/logs.go [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
+			for _, provider := range helmitObject.Replicas {
+					fmt.Println("--------------------------------------------------------------------------------------------------------------")
+					fmt.Println("--------------------------------------- Host " + provider.Host)
+					fmt.Println("--------------------------------------------------------------------------------------------------------------")
+					for _, container := range provider.Containers {
+					  	fmt.Println("--------------------------------------------------------------------------------------------------------------")
+						  fmt.Println("--------------------------------------------------------------------------------------------------------------")
+						  fmt.Println("--------------------------------------- Logs For " + container.Name)
+					    fmt.Println("--------------------------------------------------------------------------------------------------------------")
+							fmt.Println("--------------------------------------------------------------------------------------------------------------")
```
Comment (jkurz) [inline cmd/logs.go]: I don't know why the spacing looks like this. It does not look this way in atom.

**Comment by jkurz** [comment]
example output:

```
C02J41R5DKQ4:hello-harbor-compose jkurz$ harbor-compose logs
Logs For:  mss-harbor-app dev
2016/07/06 14:57:39 http://helmit.services.dmtio.net/harbor/corp-prod/mss-harbor-app/dev
--------------------------------------------------------------------------------------------------------------
--------------------------------------- Host 10.63.13.239
--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------
--------------------------------------- Logs For mss-harbor
--------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------
[2016-07-06T17:59:17.989321415Z 
 2016-07-06T17:59:17.989372160Z > mss-harbor@1.5.1 start /opt/app/current/code
 2016-07-06T17:59:17.989382426Z > node app.js
 2016-07-06T17:59:17.989390668Z 
 2016-07-06T17:59:18.338167134Z Harbor API running on 80
 2016-07-06T17:59:18.430429138Z 2016-07-06T17:59:18.429Z - info: Fetching RSS Feed (http code: 200)
 2016-07-06T18:00:20.441567405Z 2016-07-06T18:00:20.441Z - info: GET / HTTP/1.1 200 3ms
 2016-07-06T18:00:20.643885027Z Creating dynamic config
 2016-07-06T18:00:20.780447525Z 2016-07-06T18:00:20.780Z - info: GET /api/v1/blog-feed HTTP/1.1 200 71ms
 2016-07-06T18:00:36.026526852Z 2016-07-06T18:00:36.026Z - info: GET /api/v1/auth/groups/wwise HTTP/1.1 200 192ms
 2016-07-06T18:00:38.238214208Z 2016-07-06T18:00:38.237Z - info: GET /api/v1/auth/users HTTP/1.1 200 167ms
 2016-07-06T18:25:06.813879324Z 2016-07-06T18:25:06.813Z - info: GET / HTTP/1.1 200 2ms
 2016-07-06T18:25:07.017600082Z Creating dynamic config
 2016-07-06T18:25:07.302743355Z 2016-07-06T18:25:07.302Z - info: GET /api/v1/blog-feed HTTP/1.1 200 154ms
 2016-07-06T18:25:09.195748150Z 2016-07-06T18:25:09.195Z - info: GET /api/v1/auth/groups/jritsema HTTP/1.1 304 226ms
 2016-07-06T18:25:17.998617226Z 2016-07-06T18:25:17.998Z - info: GET /api/v1/auth/users HTTP/1.1 200 169ms
 2016-07-06T18:25:43.334360471Z 2016-07-06T18:25:43.333Z - info: creating datadog embed Elb Latency (last 4 hours)
 2016-07-06T18:25:43.338014716Z 2016-07-06T18:25:43.337Z - info: creating datadog embed Elb Request Count (last 4 hours)
 2016-07-06T18:25:43.339567138Z 2016-07-06T18:25:43.339Z - info: creating datadog embed CPU Usage (last 4 hours)
```

File: cmd/logs.go [hunk 2]
```diff
@@ -1,21 +1,49 @@
...
+		for shipmentName, shipment := range harborCompose.Shipments {
+			fmt.Println("Logs For:  " + shipmentName + " " + shipment.Env)
+			helmitObject := HelmitResponse{}
+			var response = GetLogs(shipment.Barge, shipmentName, shipment.Env)
+			err := json.Unmarshal([]byte(response), &helmitObject)
+			if err != nil {
+			    fmt.Println(err)
+			}
+			for _, provider := range helmitObject.Replicas {
+					fmt.Println("--------------------------------------------------------------------------------------------------------------")
```
Comment (wilsonodk) [inline cmd/logs.go]: Might be nicer if these lines were a little shorter

```
----------
--- Host:  aaa.bbb.ccc.ddd
----------
--- Name:  foo-bar-app
--- Id:    asdfghjkl
--- Image: lorum-ipsum
----------
```

As is, a small terminal window will really bork the display

File: cmd/logs.go [hunk 3]
```diff
@@ -1,21 +1,49 @@
...
+					fmt.Println("--------------------------------------- Host " + provider.Host)
+					fmt.Println("--------------------------------------------------------------------------------------------------------------")
+					for _, container := range provider.Containers {
+					  	fmt.Println("--------------------------------------------------------------------------------------------------------------")
+						  fmt.Println("--------------------------------------------------------------------------------------------------------------")
+						  fmt.Println("--------------------------------------- Name: " + container.Name)
```
Comment (wilsonodk) [inline cmd/logs.go]: `fmt.Printf` is your friend, use him…

`fmt.Printf("--- Name: %s", container.Name)`

File: cmd/harborAPI.go [hunk 4]
```diff
@@ -129,6 +130,27 @@ func update(token string, url string, data interface{}) {
 	}
 }
 
+// GetLogs returns all logs for a shipment
+// TODO: add a tail flag feature (jkurz)
```
Comment (jritsema) [inline cmd/harborAPI.go]: Ideally, we should try to mimic the [docker-compose `logs` command](https://docs.docker.com/compose/reference/logs/).

Comment (jkurz) [inline cmd/harborAPI.go]: good point. I'll play around with docker-compose logs more and implement features more like they do.

**Comment by jkurz** [comment]
now the output looks like 

```
C02J41R5DKQ4:hello-harbor-compose jkurz$ harbor-compose logs
Logs For:  mss-docker-hello-world prod
2016/07/06 18:39:01 http://helmit.services.dmtio.net/harbor/prod/mss-docker-hello-world/prod
--- Name: docker-hello-world
--- Id: cbd2cc76e19077254622b9904fa4d607173155e643d2c91f39f025a5829a6078
--- Image jkurz/hello-world:0.1.0
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/NetworkModerateUserModule/NetworkModerateUserModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/NetworkResultContentModule/NetworkResultContentModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/NetworkResultUserModule/NetworkResultUserModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/NetworksDirectoryModule/NetworksDirectoryModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/NewestGroupsModule/NewestGroupsModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/PeopleModule/PeopleModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/PopularTagsModule/PopularTagsModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68694 open() "/app/webroot/web/BetaBlockModules/PostContentModule/PostContentModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/ProfileFeedModule/ProfileFeedModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/RecentCommentsModule/RecentCommentsModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/RecentPostModule/RecentPostModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/RecentTagsModule/RecentTagsModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/RegisterModule/RegisterModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/SearchGroupsModule/SearchGroupsModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/ShowAnnouncementModule/ShowAnnouncementModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/ShowContentModule/ShowContentModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/TakerATourModule/TakerATourModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/UploadMediaModule/UploadMediaModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/UserMessagesModule/UserMessagesModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/UserPhotoModule/UserPhotoModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/VideosMediaGalleryModule/VideosMediaGalleryModule.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/BetaBlockModules/ViewAllMembersModule/ViewAllMembersModule.php" failed (2: No such file or directory)
client: 172.17.71.0
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/Flickrclient.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/help.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/includes/blogger.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/includes/functions/auto_email_notify.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/includes/functions/html_generate.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/includes/functions/validations.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/index.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/lib/xml/oai/ListRecords.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/login.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/logout.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/lom.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/network_module_selector.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/submit_abuse.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/web/submit_comment.php" failed (2: No such file or directory)
2016/07/06 16:56:22 [error] 8#0: *68695 open() "/app/webroot/webavis/class/class.php" failed (2: No such file or directory)
```

File: cmd/harborAPI.go [hunk 5]
```diff
@@ -129,6 +130,26 @@ func update(token string, url string, data interface{}) {
 	}
 }
 
+// GetLogs returns a string of all container logs for a shipment
+func GetLogs(barge string, shipment string, env string) string {
+    var url string = helmitURI + "/harbor/" + barge + "/" + shipment + "/" + env
```
Comment (jritsema) [inline cmd/harborAPI.go]: Should we use the 'uritemplates' library to be consistent with the other functions in this file?

File: cmd/harborAPI.go [hunk 6]
```diff
@@ -129,6 +130,26 @@ func update(token string, url string, data interface{}) {
...
+
+		log.Println(url)
```
Comment (jritsema) [inline cmd/harborAPI.go]: I've been checking for the --verbose flag before writing out debug/extraneous type info...

``` go
if Verbose {
  log.Println("extra info")
}
```

**Pull request merged**